webui/react/src/shared/components/ActionDropdown/ActionDropdown.tsx

```diff
@@ -18,17 +18,23 @@ export type Confirmations<T extends string> =
   Partial<{ [key in T]: Omit<ModalFuncProps, 'onOk'> }>
 type DisabledActions<T extends string> =
   Partial<{ [key in T]: boolean }>
+type DangerousActions<T extends string> = DisabledActions<T>;
 
 interface Props<T extends string> {
   /**
    * define the order of actions to show up in the dropdown menu.
    */
   actionOrder: T[];
+  children?: React.ReactNode;
   /**
    * whether to prompt the user to confirm the action before executing it
    * with options to customize the generated modal.
    */
   confirmations?: Confirmations<T>
+  /**
+   * whether the action is marked as dangerous or not.
+   */
+  danger?: DangerousActions<T>;
```

Comment:
Not necessarily in this PR but I was thinking it might be nice to reuse the Action interface in TableBatch for this as well. 

That way we can just add confirmations, dangerous, and the like to the Action interface themselves instead of having these extra metadata props. 

What do you think?